pkg/services/baremetal/client/robot/robot_client.go

```diff
@@ -45,10 +50,32 @@ type Factory interface {
 	NewClient(Credentials) Client
 }
 
+type LoggingTransport struct {
+	http.RoundTripper
+}
+
+var replaceHex = regexp.MustCompile(`0x[0123456789abcdef]+`)
+
+func (lt *LoggingTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+
+	stack := replaceHex.ReplaceAllString(string(debug.Stack()), "0xX")
+
+	resp, err = lt.RoundTripper.RoundTrip(req)
+	if err != nil {
+		log.Printf("hetzner robot API. Error: %s %4s %s %q", err, req.Method, req.URL, stack)
```

Comment:
use https://github.com/syself/cluster-api-provider-hetzner/blob/main/pkg/utils/utils.go#L195 with whatever logLevel you want